src/api/RestAPI.ts

```diff
@@ -312,7 +314,12 @@ export class RestAPI extends EventEmitter {
                 }),
                 // Repeater
                 (async function repeater(): Promise<A> {
-                    const result = await promise();
+                    const result = await new Promise<A>((resolve) => {
+                        setTimeout(async () => {
+                            const r = await promise();
+                            resolve(r);
+                        }, interval);
+                    });
```

Comment:
Not important, feel free to skip this comment.

This could be simplified to:

```tsx
await new Promise(resolve => setTimeout(resolve, interval)); // sleep to avoid firing requests too fast
const result = await promise();
```

Or am I missing something? 🙂

Reply:
Also it may be overkilling it but we may want to add a flag in `execRequest` to know if this is the first time and do the first trigger immediately. Quite a lot of poll end up only doing one call on the transaction history list and we do not want to make it slower in those cases 🙂

Reply:
let me see if i can refactor the execRequest logic too